### PR TITLE
MBeans

### PR DESCRIPTION
**Pull request by vasilmkd**
Finally. Only 8 months late.

### CODE REVIEW

**Comment by vasilmkd** [comment]
Screen Shot 2021-11-17 at 18 02 21

**Comment by vasilmkd** [comment]
You have Cats Effect 3.3.0.

**Review by djspiewak — DISMISSED** [review]
LOL. Nice work. :-)

File: core/shared/src/main/scala/cats/effect/unsafe/IORuntime.scala [hunk 1]
```diff
@@ -36,13 +36,20 @@ final class IORuntime private (
...
 
   private[effect] val fiberMonitor: FiberMonitor = FiberMonitor(compute)
 
+  private[this] val unregisterMBean = registerFiberMonitorMBean(fiberMonitor)
+
+  val shutdown: () => Unit = { () =>
+    unregisterMBean()
+    _shutdown()
+  }
+
```
Comment (vasilmkd) [inline core/shared/src/main/scala/cats/effect/unsafe/IORuntime.scala]: It's horrible that this code is in the constructor, let me see if I can do something about it.

**Comment by vasilmkd** [comment]
Screen Shot 2021-11-17 at 18 45 03

Still here.

File: core/shared/src/main/scala/cats/effect/unsafe/IORuntime.scala (hunk 1 above)
Comment (vasilmkd) [reply core/shared/src/main/scala/cats/effect/unsafe/IORuntime.scala]: Done.

**Comment by vasilmkd** [comment]
Ok, this is now not the same as before. If the registration code is in the constructor, every `IORuntime` constructed would register the MBean. Otherwise, only the `IOApp` and `global` ones do.

@djspiewak Thoughts?

I can easily revert the last commit.

**Comment by djspiewak** [comment]
@vasilmkd IMO it's better to just do it in `IOApp` and `global`. Better not to mess with random third-party runtimes.

**Comment by vasilmkd** [comment]
Ok, ready to go then.

Edit: See questions below.

**Comment by vasilmkd** [comment]
Actually, another question. Should we do that for `WSTP`s as well?

**Comment by vasilmkd** [comment]
At the moment, every `WSTP` constructed registers metrics. Should we also tie it to `IOApp` and `global` only?

**Comment by djspiewak** [comment]
> At the moment, every WSTP constructed registers metrics. Should we also tie it to IOApp and global only?

I think tying it to `IOApp` and `global` is better in general, though maybe this is academic since you can't actually create a WSTP from third-party code.

**Comment by armanbilge** [comment]
> though maybe this is academic since you can't actually create a WSTP from third-party code.

Ah, you can't. I was mildly surprised since for a moment there it seemed like you could :) I think we can merge this PR as is.

**Review by djspiewak — APPROVED** [review]

**Pull request merged**